src/components/user-profile-info/UserProfileInfo.tsx

```diff
@@ -50,10 +50,10 @@ const UserProfileInfo: FC<UserProfileInfoProps> = ({
   sx = {},
   _id,
   role,
-  renderAdditionalInfo = true,
-  onlineBadge = false
+  renderAdditionalInfo = true
 }) => {
   const { t } = useTranslation()
+  const { usersOnline } = useAppSelector((state) => state.socket)
```

Comment:
Consider doing this in the parent component, so every `usersOnline` change doesn't trigger re-render

Reply:
To optimise re-renders instead of moving `useAppSelector` to the parent component I decided to create memoized selector that only checks if the specific user is online. This will only trigger re-renders for the component when the online status of the specific user (id) changes, instead of whenever the `usersOnline` array is modified. This selector I put in the `redux/selectors` folder. As a result I managed to optimise re-renders of `UserProfileInfo`, `ChatItem` and `ChatHeader` components in my latest commit.